addons/hr_expense/hr_expense.py

```diff
@@ -64,94 +56,85 @@ def _get_currency(self, cr, uid, context=None):
-        if journal_ids:
-            res['value']['journal_id'] = journal_ids[0]
-        return res
-
-    def onchange_employee_id(self, cr, uid, ids, employee_id, context=None):
-        emp_obj = self.pool.get('hr.employee')
-        department_id = False
-        employee_payable_account_id = False
-        company_id = False
-        if employee_id:
-            employee = emp_obj.browse(cr, uid, employee_id, context=context)
-            department_id = employee.department_id.id
-            company_id = employee.company_id.id
-            if employee.address_home_id and employee.address_home_id.property_account_payable:
-                employee_payable_account_id = employee.address_home_id.property_account_payable.id
-        return {'value': {'department_id': department_id, 'company_id': company_id, 'employee_payable_account_id': employee_payable_account_id}}
-
-    def expense_confirm(self, cr, uid, ids, context=None):
-        for expense in self.browse(cr, uid, ids):
-            if not expense.line_ids:
-                raise osv.except_osv(_('Error!'), _('You cannot submit expense which has no expense line.'))
-            if expense.employee_id and expense.employee_id.parent_id.user_id:
-                self.message_subscribe_users(cr, uid, [expense.id], user_ids=[expense.employee_id.parent_id.user_id.id])
-        return self.write(cr, uid, ids, {'state': 'confirm', 'date_confirm': time.strftime('%Y-%m-%d')}, context=context)
-
-    def expense_accept(self, cr, uid, ids, context=None):
-        return self.write(cr, uid, ids, {'state': 'accepted', 'date_valid': time.strftime('%Y-%m-%d'), 'user_valid': uid}, context=context)
-
-    def expense_canceled(self, cr, uid, ids, context=None):
-        return self.write(cr, uid, ids, {'state': 'cancelled'}, context=context)
-
-    def account_move_get(self, cr, uid, expense_id, context=None):
+    name = fields.Char(string='Expense Sheet', readonly=True, states={'draft':[('readonly', False)], 'confirm':[('readonly', False)]})
+    date = fields.Date(string='Date', select=True, readonly=True, states={'draft':[('readonly', False)], 'confirm':[('readonly', False)]}, default=fields.Date.context_today)
+    journal_id = fields.Many2one('account.journal', string='Force Journal', help = "The journal used when the expense is done.")
+    employee_payable_account_id = fields.Many2one('account.account', string='Employee Account', help="Employee payable account")
+    employee_id = fields.Many2one('hr.employee', string="Employee", required=True, readonly=True, states={'draft':[('readonly', False)], 'confirm':[('readonly', False)]}, default=_employee_get)
+    user_id = fields.Many2one('res.users', string='User', required=True, default=lambda self: self.env.user)
+    date_confirm = fields.Date(string='Confirmation Date', select=True, copy=False,
```

Comment:
@rim-odoo 
date_confirm, date_valid, fields used on postgres report is find difference beetween "Delay to Confirmed" or "Delay to validate" Employee sheet .